database/mongodb/README.md

```diff
@@ -0,0 +1,19 @@
+# MongoDB
+
+* Driver work with mongo through [db.runCommands](https://docs.mongodb.com/manual/reference/command/)
+* Migrations support json format. It contains array of commands for `db.runCommand`. Every command is executed in separate request to database 
```

Comment:
Is it possible to start a transaction/session in a migration?

e.g. https://docs.mongodb.com/manual/reference/command/startSession/#dbcmd.startSession

If so, can you add a test that uses a transaction/session? I'm not sure how this would be managed in a migration. It may be cleaner to add an option for the driver to wrap the migration with a transaction... If that's the case, I think it's fine to defer that work, but we should make a note of that in the README.

Reply:
I'm not familiar with running a MongoDB cluster, much less inside Docker.

Is there a way to run the nodes in a single container? Or do we need to use multiple containers?

We shouldn't need to run a container in interactive mode.
The [MongoDB Docker docs](https://hub.docker.com/_/mongo/) suggest that we can pass in parameters to the `mongod` command. I believe this is your 2nd approach, but IDK if we can run all of the nodes/replicas in a single container. The `migrate` docker testing framework currently doesn't support testing against multiple containers.

I wouldn't bother adding such a feature since I'm thinking about moving the testing framework to [my own](https://github.com/dhui/dktest), but it's not quite a drop in replacement. It may be easier to make the necessary changes to `dktest` and integrate it into `migrate`.

If adding tests for MongoDB transactions turns out to be too complicated, we can defer this work for now as long as we properly document the lack of stability.

Reply:
> Is there a way to run the nodes in a single container? Or do we need to use multiple containers?

We can run only one mongod node and initiate replica it's gonna be one primary node.
```sh
mongod --port 27017 --bind_ip_all --replSet rs0
```
Then throug the docker SDK need to execute command
```sh
docker exec $MONGO_CONTAINER mongo --eval 'rs.initiate()'
```
It's enough for testing.


> We shouldn't need to run a container in interactive mode.

We should run it only if we decide chose 1st approach. For shell it's gonna like:
```sh
docker run -id -P --name mongodb mongo:4 bin/bash -c "mongod --fork --logpath /data/log.log --port 27017 --replSet rs0 && mongo --eval 'rs.initiate()' && bash"
```

Reply:
It's probably cleaner to run each MongoDB node in its own container.
We may run into issues with the nodes clobbering each other when run in the same container. e.g. binding to the same port, ready from the same DB path, log paths, etc

We'll probably have to do something similar to [docker-compose](https://docs.docker.com/compose/overview/)

Reply:
With some tricks, i did it. Look at last commit, there is test for migrations in transaction.